client/actions/actions.js

```diff
@@ -0,0 +1,137 @@
+import * as ActionTypes from './actionTypes';
+import { asyncAction } from '../middleware/asyncMiddleware';
+import axios from 'axios';
+
+/**
+ * Action Creators
+ */
+
+export const createUser = (email, password) => {
+    return asyncAction(axios.post('/users', { email, password })
+            .then(response => response.data)
+            .then(createUserSuccessActions)
+            .catch(createUserFailure));
+};
+
+const createUserSuccessActions = data => {
+    return dispatch => {
+        dispatch(createUserSuccess());
+        dispatch(login(data));
+    }
+};
```

Comment:
This action is actually a function that returns a function, essentially all it's doing is dispatching two actions in a row. On it's own, Redux can't do this, it needs a middleware like `redux-thunk` to work.